File: apis/src/main/java/com/beat/apis/home/facade/HomeFacade.java
package com.beat.apis.home.facade;

import org.springframework.stereotype.Service;

import com.beat.apis.home.application.HomeService;
import com.beat.apis.home.application.dto.HomeFindAllResponse;
import com.beat.apis.home.application.dto.HomeFindRequest;
import com.beat.apis.home.application.dto.HomeGenreType;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class HomeFacade {
	private final HomeService homeService;

	public HomeFindAllResponse findHomePerformanceList(HomeGenreType genre) {
		return homeService.findHomePerformanceList(new HomeFindRequest(genre));
	}
}

File: apis/src/main/kotlin/com/beat/apis/home/facade/HomeFacade.kt
package com.beat.apis.home.facade

import com.beat.apis.home.api.response.HomeFindAllResponse
import com.beat.apis.home.api.type.HomeGenreType
import com.beat.application.frontoffice.home.booker.query.HomeQueryService
import org.springframework.stereotype.Service

@Service
class HomeFacade(
    private val homeQueryService: HomeQueryService,
) {
    fun findHomePerformanceList(genre: HomeGenreType?): HomeFindAllResponse =
        HomeFindAllResponse.from(homeQueryService.findHomePerformanceList(genre?.name))
}
